docs/Lexicon.md

```diff
@@ -0,0 +1,562 @@
+guarantee this. This was 
+[discussed on Apple's Swift blog](https://developer.apple.com/swift/blog/?id=28).
+
+## DNM
+
+"Do not merge". Placed in PR titles where discussion or analysis is still
+ongoing.
+
+## dup
+
+From "duplicate". As a noun, refers to another filed issue that describes
+the same bug ("I have a dup of this"); as a verb, the act of marking a bug
+*as* a duplicate ("Please dup this to the underlying issue"). Sometimes
+written "dupe". Pronounced the same way as the first syllable of
+"duplicate", which for most American English speakers is "doop".
+
+## existential
+
+A value whose type is a protocol composition (including a single protocol
+and *zero* protocols; the latter is the `Any` type).
+
+## fragile
+
+Describes a type or function where making changes will break binary
+compatibility. See [LibraryEvolution.rst](LibraryEvolution.rst).
+
+## generic environment
+
+Provides context for interpreting a type that may have generic parameters
+in it. Generic parameter types are normally just represented as "first
+generic parameter in the outermost context" (or similar), so it's up to the
+generic environment to note that that type must be a Collection. (Another
+way of looking at it is that the generic environment connects
+[interface types](#interface-type) with
+[contextual types](#contextual-type)).
+
+## generic signature
+
+A representation of all generic parameters and their requirements. Like
+types, generic signatures can be [canonicalized](#canonical-type) to be
+compared directly.
+
+## iff
+
+["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if). This term comes from mathematics.
+
+## interface type
+
+The type of a value or declaration outside its generic context. These types
+are written using "formal" generic types, which only have meaning when
+combined with a particular generic declaration's "generic signature".
+Unlike [contextual types](#contextual-type), interface types store
+conformances and requirements in the generic signature and not in the types
+themselves. They can be compared across declarations but cannot be used
+directly from within the context.
+
+## irrefutable pattern
+
+A pattern that always matches. These patterns either bind to a variable or
+perform structural modification, e.x.:
+
+1. `case _:`.
+2. `case let x:`.
+3. `case (_, _):`.
+
+## IR
+
+1. "intermediate representation": a generic term for a format representing
+   code in a way that is easy for a compiler or tool to manipulate.
+2. "LLVM IR": a particular IR used by the LLVM libraries for optimization
+   and generation of machine code.
+
+## IUO (implicitly unwrapped optional)
+
+A type like Optional, but it implicitly converts to its wrapped type. If
+the value is `nil` during such a conversion, the program traps just as
+it would when a normal Optional is force-unwrapped. IUOs implicitly
+convert to and from normal Optionals with the same wrapped type.
```

Comment:
Outdated? [*Reimplementation of Implicitly Unwrapped Optionals*](https://swift.org/blog/iuo/)